// proto/pdf_service.proto
syntax = "proto3";

package pdf_service;

service PDFService {
  // Process and store PDF
  rpc ProcessPDF (ProcessPDFRequest) returns (ProcessPDFResponse) {}
  
  // Query PDF content
  rpc QueryPDF (QueryPDFRequest) returns (QueryPDFResponse) {}
  
  // Stream query PDF content
  rpc StreamQueryPDF (QueryPDFRequest) returns (stream StreamQueryPDFResponse) {}
  
  // Get PDF metadata
  rpc GetPDFMetadata (GetPDFMetadataRequest) returns (GetPDFMetadataResponse) {}
}

message ProcessPDFRequest {
  bytes pdf_content = 1;
  string filename = 2;
}

message ProcessPDFResponse {
  bool success = 1;
  string error_message = 2;
}

message QueryPDFRequest {
  string query = 1;
  string pdf_id = 2;
}

message QueryPDFResponse {
  string answer = 1;
  bool success = 2;
  string error_message = 3;
}

message StreamQueryPDFResponse {
  string chunk = 1;
  bool success = 2;
  string error_message = 3;
}

message GetPDFMetadataRequest {
  string pdf_id = 1;
}

message GetPDFMetadataResponse {
  string title = 1;
  string author = 2;
  int32 page_count = 3;
  bool success = 4;
  string error_message = 5;
}